six/modules/python/six/tests/runTests.py

```diff
@@ -70,31 +70,36 @@ def runCPHDTests(cphdDir):
     # Directory of CPHD1.0 schemas
     cphdSchemaDir = os.path.join(utils.findSixHome(), 'six',
                                  'modules', 'c++', 'cphd', 'conf', 'schema')
-    cphdSchemas = []
-    for schema in os.listdir(cphdSchemaDir):
-        cphdSchemas.append(os.path.join(cphdSchemaDir, schema))
 
+    # Init executable path
     os.environ['PATH'] = (os.environ['PATH'] + os.pathsep +
                           os.path.join(utils.installPath(), 'tests', 'cphd'))
     writeCphd = utils.executableName('test_round_trip')
-    cphdPathname = os.path.join(cphdDir, os.listdir(cphdDir)[0])
 
-    # Generate CPHD1.0 output file
-    success = subprocess.call([writeCphd,
-                               cphdPathname, 'output.cphd', cphdSchemas[0]],
-                              stdout=subprocess.PIPE)
+    # Run cphd test for each CPHD version
+    cphdSchemas = []
+    result = True
+    for i in range(len(os.listdir(cphdSchemaDir))):
+        cphdSchemas.append(os.path.join(cphdSchemaDir, os.listdir(cphdSchemaDir)[i]))
 
-    print('Running test_round_trip')
+        cphdPathname = os.path.join(cphdDir, os.listdir(cphdDir)[i])
 
-    if success != 0:
-        if os.path.exists('output.cphd'):
-            os.remove('output.cphd')
-        print('Error running test_round_trip')
-        return False
+        # Generate CPHD1.0 output file
+        success = subprocess.call([writeCphd,
```

Comment:
If I'm understanding this right, you're pairing each CPHD in the test directory with a schema of the same version, and doing a round-trip/compare.

It's not valid to assume that `os.listdir` will return files in a certain order. I *think* the xercesc/xml.lite layer is smart enough to find the appropriate schema if you give it the directory containing the schemas.